### PR TITLE
Fix release pipeline issues for the next release

### PR DESCRIPTION
**Pull request by skyline75489**
Preparing for 0.5.0 release:

* Drop Python 3.8 & 3.9 support
* Add Python 3.13 support.
* Remove `ort-nightly` across pipelines.

---

NuGet validation run:
* https://aiinfra.visualstudio.com/ONNX%20Runtime/_build/results?buildId=592239&view=results

Python validation run:
* https://aiinfra.visualstudio.com/ONNX%20Runtime/_build/results?buildId=591237&view=results

### CODE REVIEW

File: .pipelines/pypl-publishing.yml [hunk 1]
```diff
@@ -37,38 +37,13 @@ parameters:
 - name: enable_linux_rocm
   displayName: 'Whether Linux ROCm package is built.'
   type: boolean
-  default: true
+  default: false
```
Comment (skyline75489) [inline .pipelines/pypl-publishing.yml:40]: ROCm is not ready

File: .pipelines/pypl-publishing.yml [hunk 2]
```diff
@@ -37,38 +37,13 @@ parameters:
...
 
 - name: enable_macos_cpu
   displayName: 'Whether MacOS CPU package is built.'
   type: boolean
   default: true
 
-- name: ort_version
```
Comment (skyline75489) [inline .pipelines/pypl-publishing.yml:47]: These versions are no long used after @baijumeswani 's previous work. The ORT downloads are handled in cmake.

File: .pipelines/stages/jobs/steps/capi-win-step.yml [hunk 3]
```diff
@@ -26,38 +26,20 @@ steps:
...
     echo "arch=$(arch)"
-    echo "ort_filename=$(ort_filename)"
     echo "ort_version=$(ort_version)"
     echo "ep=$(ep)"
     echo "cuda_version=$(cuda_version)"
     echo "target=${{ parameters.target }}"
     echo "build_config=${{ parameters.build_config }}"
   displayName: 'Print Parameters'
 
-- ${{ if eq(parameters.ep, 'directml') }}:
```
Comment (skyline75489) [inline .pipelines/stages/jobs/steps/capi-win-step.yml:43]: No longer use. Handled in cmake.

File: cmake/package.cmake [hunk 4]
```diff
@@ -14,6 +14,14 @@ install(TARGETS
   PUBLIC_HEADER DESTINATION include
   FRAMEWORK DESTINATION ${CMAKE_INSTALL_BINDIR}
 )
+if(USE_CUDA)
```
Comment (skyline75489) [inline cmake/package.cmake:17]: This is to pick up the `onnxruntime-genai-cuda.dll` after @RyanUnderhill 's work.

File: src/python/setup.py.in [hunk 5]
```diff
@@ -26,22 +26,21 @@ package_name = '@TARGET_NAME@'
 def _onnxruntime_dependency() -> str:
     dependency = None
     # Use stable version as default
-    ort_version = os.environ.get("ONNXRUNTIME_VERSION", "1.19.0")
-    is_nightly = True if "dev" in ort_version else False
```
Comment (skyline75489) [inline src/python/setup.py.in:30]: ort-nightly is gone. i think not a lot people will miss it.

**Review by aciddelgado — APPROVED** [review]

**Pull request merged**